src/TableMetaData.java

```diff
@@ -0,0 +1,131 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.io.RandomAccessFile;
+import java.math.BigInteger;
+
+/* This class is used to read and manipulate the meta data of the table (davisbase_tables and davisbas_columns)
+    we need to make sure to update the meta data
+        - record_count
+        - root page No
+     of the table whenever a records gets inserted or deleted
+       
+*/
+public class TableMetaData{
+
+    public int recordCount;
+    public List<TableRecord> columnData;
+    public List<ColumnInfo> columnNameAttrs;
+    public List<String> columnNames;
+    public String tableName;
+    public boolean tableExists;
+    public int rootPageNo;
+
+    public TableMetaData(String tableName)
+    {
+        this.tableName = tableName;
+        tableExists = false;
+        try {
+
+            RandomAccessFile davisbaseTablesCatalog = new RandomAccessFile(
+                DavisBaseBinaryFile.getDataFilePath(DavisBaseBinaryFile.tablesTable), "r");
+            
+            //get the root page of the table
+            int rootPageNo = DavisBaseBinaryFile.getRootPageNo(davisbaseTablesCatalog);
+           
+            BPlusOneTree bplusOneTree = new BPlusOneTree(davisbaseTablesCatalog, rootPageNo);
+            //search through all leaf papges in davisbase_tables
+            for (Integer pageNo : bplusOneTree.getAllLeaves()) {
+               Page page = new Page(davisbaseTablesCatalog, pageNo);
+               //search theough all the records in each page
+               for (TableRecord record : page.records) {
+                   //if the record with table is found, get the root page No and record count; break the loop
+                  if (new String(record.getAttributes().get(0).fieldValue).equals(tableName)) {
+                    rootPageNo = Integer.parseInt(record.getAttributes().get(3).fieldValue);
+                    recordCount = Integer.parseInt(record.getAttributes().get(2).fieldValue);
+                    tableExists = true;
+                     break;
+                  }
+               }
+               if(tableExists)
+                break;
+            }
+   
+            davisbaseTablesCatalog.close();
+            if(tableExists)
+            {
+               loadColumnData();
+            }
```

Comment:
TODO: handle when the table does not exist - maybe return null or throw exception (ie: TableNonExistentException)